core/state/trie_prefetcher.go

```diff
@@ -332,6 +330,29 @@ func (sf *subfetcher) loop() {
 		}
 		sf.trie = trie
 	}
+	handleTask := func(task []byte) {
```

Comment:
Is this just from upstream?

Reply:
I moved this from https://github.com/ava-labs/coreth/pull/666/files/7b9959050f89377e30b15957fd5081e012cd720c..96575b96ada7cc778f0a53c5045a9539f5ec530b#diff-58140c10803d5c33def41482e87d096548e88382aacc21aa7dbee1e918915870L361-L370 so I can call it in defer.

Using defer seems cleaner than to modify the channel logic in the loop.
As we would encounter a merge conflict for the code that handles the task since we modified GetAccount to PrefetchAccount  (to avoid wrapping GetAccount)

Here's another approach for example https://github.com/ava-labs/subnet-evm/commit/5a48a7d358e3cbe621f6fb7455826c635e193f40#diff-58140c10803d5c33def41482e87d096548e88382aacc21aa7dbee1e918915870R361 (but this does not support the original pattern).